daemon/emer-daemon.c

```diff
@@ -564,21 +593,12 @@ handle_http_response (SoupSession *http_session,
       SOUP_STATUS_IS_CLIENT_ERROR (status_code) ||
       SOUP_STATUS_IS_SERVER_ERROR (status_code))
     {
-      backoff (priv->rand, callback_data->attempt_num);
-
-      GError *error = NULL;
-      GVariant *updated_request_body =
-        get_updated_request_body (self, callback_data->request_body, &error);
-      if (updated_request_body == NULL)
-        {
-          g_task_return_error (upload_task, error);
-          finish_network_callback (upload_task);
-          return;
-        }
-
-      g_variant_unref (callback_data->request_body);
-      callback_data->request_body = updated_request_body;
-      queue_http_request (upload_task);
+      guint random_backoff_interval =
+        get_random_backoff_interval (priv->rand, callback_data->attempt_num);
+      callback_data->backoff_timeout_source_id =
+        g_timeout_add (random_backoff_interval,
```

Comment:
Given that this only happens in error conditions that should not be common, I don't think the impact on power consumption (the main reason to use `g_timeout_add_seconds()` AFAICS) would be noticeable... But you're right, technically we don't need millisecond precision here, so I will change it.